dnf/module/repo_module_dict.py

```diff
@@ -0,0 +1,761 @@
+
+        repo_module = module_version.repo_module
+
+        if repo_module.conf.locked._get():
+            raise VersionLockedException(module_spec, module_version.version)
+
+        repo_module.disable()
+
+        if save_immediately:
+            self.base._module_persistor.commit()
+            self.base._module_persistor.save()
+
+    def lock(self, module_spec, save_immediately=False):
+        subj = ModuleSubject(module_spec)
+        module_version, module_form = subj.find_module_version(self)
+
+        repo_module = module_version.repo_module
+
+        if not repo_module.conf.enabled._get():
+            raise EnabledStreamException(module_spec)
+        elif repo_module.conf.locked._get() and \
+                (repo_module.conf.stream._get() != module_version.stream or
+                 repo_module.conf.version._get() != module_version.version):
+            raise VersionLockedException(module_spec, module_version.version)
+
+        version_to_lock = module_version.version
+        if list(repo_module.conf.profiles._get()):
+            version_to_lock = module_version.repo_module.conf.version._get()
+        repo_module.lock(version_to_lock)
+
+        if module_form.version and version_to_lock != module_form.version:
+            raise CannotLockVersionException(module_spec, module_form.version,
+                                             "Different version installed.")
+
+        if save_immediately:
+            self.base._module_persistor.commit()
+            self.base._module_persistor.save()
+
+        return module_version.stream, version_to_lock
+
+    def unlock(self, module_spec, save_immediately=False):
+        subj = ModuleSubject(module_spec)
+        module_version, module_form = subj.find_module_version(self)
+
+        repo_module = module_version.repo_module
+
+        if not repo_module.conf.enabled._get():
+            raise EnabledStreamException(module_spec)
+
+        repo_module.unlock()
+
+        if save_immediately:
+            self.base._module_persistor.commit()
+            self.base._module_persistor.save()
+
+        return module_version.stream, module_version.version
+
+    def install(self, module_specs):
+        versions, module_specs = self.get_best_versions(module_specs)
+
+        result = False
+        for module_version, profiles, default_profiles in versions.values():
+            conf = module_version.repo_module.conf
+            if conf.locked._get() and conf.version._get() != module_version.version:
+                logger.warning(module_messages[VERSION_LOCKED]
+                               .format(module_version.name,
+                                       module_version.repo_module.conf.version._get()))
+                continue
+
+            self.enable("{}:{}".format(module_version.name, module_version.stream))
+
+        self.base.sack.reset_includes()
+        self.base.sack.reset_excludes()
+        self.base._setup_excludes_includes()
```

Comment:
I think the previous 3 lines it should be replaced by:
self.base.sack.reset_module_excludes()
self.base.sack.use_module_includes()